data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -123,33 +116,53 @@ public GrokPrepper(final PluginSetting pluginSetting) {
             } catch (JsonProcessingException e) {
                 LOG.error("Failed to parse the record [{}]", record.getData());
                 recordsOut.add(record);
+            } catch (TimeoutException e) {
+                LOG.error("Matching on record [{}] took longer than [{}] and timed out", record.getData(), grokPrepperConfig.getTimeoutMillis());
+                recordsOut.add(record);
+            } catch (ExecutionException e) {
+                LOG.error("An exception occurred while matching on record [{}]", record.getData(), e);
+                e.printStackTrace();
```

Comment:
I'm guessing this came from IntelliJ's auto-complete. We should never be calling `printStackTrace()`. The line above will print it to the logger just fine.